sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveMetadataCacheSuite.scala

```diff
@@ -126,4 +129,39 @@ class HiveMetadataCacheSuite extends QueryTest with SQLTestUtils with TestHiveSi
   for (pruningEnabled <- Seq(true, false)) {
     testCaching(pruningEnabled)
   }
+
+  test("cache TTL") {
+    val sparkConfWithTTl = new SparkConf().set(SQLConf.METADATA_CACHE_TTL.key, "1")
+    val newSession = SparkSession.builder.config(sparkConfWithTTl).getOrCreate().cloneSession()
+
+    withSparkSession(newSession) { implicit spark =>
```

Comment:
ditto

Reply:
So, `withSQLConf` doesn't work here. It seems like it overrides the config, but it doesn't affect already initialized `SessionCatalog` (from the previous tests; it does work if you run it as a single test).

`withSQLConf` works in `FileIndexSuite` because it uses `SharedSparkSession` as base class, but `TestHiveSingleton` is used as a base here. 

Any suggestions? I can:

-  keep it as is
-  move it to its own separate suite, either in the same file or a separate file
-  try to use a different base for the whole suite

Reply:
Ah, is this not a runtime config? If so, `SQLConf` -> `StaticSQLConf`

Reply:
Apparently `withSparkSession` (current implementation) is not good enough, it terminates the Spark session and the next tests fail https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/124516/console

Reply:
@maropu hmm, how do I use `withSQLConf` with `StaticSQLConf `? It doesn't allow it: https://github.com/apache/spark/blob/master/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/plans/SQLHelper.scala#L50

Reply:
Its okay to use `buildStaticConf` https://github.com/apache/spark/pull/28852#discussion_r445893610

Reply:
Yea, `withSQLConf` is usd only for runtime configs, then we cannot use it for static configs. That's an issue of how-to-write-tests.

Reply:
I see that `withSQLConf ` is overridden here https://github.com/apache/spark/blob/master/sql/core/src/test/scala/org/apache/spark/sql/internal/ExecutorSideSQLConfSuite.scala#L59-L68 and it works with `StaticSQLConf`: https://github.com/apache/spark/blob/master/sql/core/src/test/scala/org/apache/spark/sql/internal/ExecutorSideSQLConfSuite.scala#L59-L68

It also works If I implement something similar in my test, but it still feels hacky. Do you think we can go with this plan? In this case, the test in the `HiveMetadataCacheSuite` should be in its own suite / class.

Reply:
How about creating `SQLContext` by yourself like this? https://github.com/apache/spark/blob/master/sql/core/src/test/scala/org/apache/spark/sql/SparkSessionBuilderSuite.scala#L114-L115

Reply:
I've decided to move the test from `HiveMetadataCacheSuite` to `SessionCatalogSuite` in https://github.com/apache/spark/pull/28852/commits/18feeb02b11407b9eea1457139a91cb40ddc40cf:

- It does cover the in-memory catalog as well, which means bigger scope for the test.
- It's easier to test it since SQLConf can be provided in the test. 

I'm going to resolve this, but feel free to leave more comments on the new test.